cmd/hubagent/workload/setup.go

```diff
@@ -119,6 +120,30 @@ func SetupControllers(ctx context.Context, mgr ctrl.Manager, config *rest.Config
 	if opts.EnableV1Beta1APIs {
 		klog.Info("Setting up clusterResourcePlacement v1beta1 controller")
 		clusterResourcePlacementControllerV1Beta1 = controller.NewController(crpControllerV1Beta1Name, controller.NamespaceKeyFunc, crpc.Reconcile, rateLimiter)
+
+		// Start up the controller.
+		//
+		// Note that, with v1alpha1 controllers, all custom ones are started up alongside the
+		// resource change detector, when its Start() method is invoked by the controller
+		// manager. This is no longer the case with v1beta1 controllers, specifically, the
+		// v1beta1 CRP controller, as it depends on its own watchers. As a result, here
+		// this controller is spun up separately.
+		//
+		// The controller must run in a separate goroutine as Run() is a blocking call.
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			err := clusterResourcePlacementControllerV1Beta1.Run(ctx, opts.ConcurrentClusterPlacementSyncs)
```

Comment:
can we run the controller in the changeDetector start() func, similar to v1alpha1 controller

Reply:
Done, as we previously discussed.